core-rust/state-manager/src/transaction/round_update_transaction.rs

```diff
@@ -169,3 +163,90 @@ impl HasRoundUpdateTransactionHash for PreparedRoundUpdateTransactionV1 {
 pub trait HasRoundUpdateTransactionHash {
     fn round_update_transaction_hash(&self) -> RoundUpdateTransactionHash;
 }
+
+/// A builder of "successful/timeout/gap rounds by leader" counter update, for metrics purposes.
+#[derive(Default)]
+pub struct LeaderRoundCountersBuilder {
+    /// Counters per validator (of the corresponding [`ValidatorIndex`]).
+    /// Implementation note:
+    /// This structure starts as an empty one (on the builder's initialization via `default()`).
+    /// Then, it is lazily-initialized by any larger-than-previously-observed [`ValidatorIndex`]
+    /// (within `update()`; see `get_counter_mut()`). The size of this vector is thus effectively
+    /// bounded by `u8::MAX` (even if validator indices found in proposal history were invalid).
+    // TODO(during review): actually it's hard to say if this is more performant than `BTreeMap<ValidatorIndex, _>`...?
```

Comment:
_(note from reviewer)_

I am really curious what reviewers think!

Reply:
I agree, " it's hard to say" :trollface: happy to go with a Vec

Reply:
100% faster, would be even faster with a preallocation.